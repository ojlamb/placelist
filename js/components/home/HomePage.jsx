import React from "react";
import Button from "@material-ui/core/Button";
import { Link } from "react-router-dom";

class HomePage extends React.Component {
  constructor(props) {
    super(props);
    this.props = props;
  }

  render() {
    return (
      <div className="HomePage">
        <div className="hero">
          <h1>Welcome to PlaceList</h1>
          <p>List, map, share and rank your spots</p>
          <Link style={{ textDecoration: "none" }} to="/places">
            <Button variant="contained" color="primary">
              Explore to the List
            </Button>
          </Link>
        </div>
      </div>
    );
  }
}

export default HomePage;
