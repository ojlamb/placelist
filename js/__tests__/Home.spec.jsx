import React from "react";
import { shallow } from "enzyme";
import HomePage from "../components/home/HomePage";

test("Home renders correctly", () => {
  const component = shallow(<HomePage />);
  expect(component).toMatchSnapshot();
});
