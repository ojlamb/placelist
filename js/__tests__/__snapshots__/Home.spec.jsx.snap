// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home renders correctly 1`] = `
<div
  className="HomePage"
>
  <div
    className="hero"
  >
    <h1>
      Welcome to PlaceList
    </h1>
    <p>
      List, map, share and rank your spots
    </p>
    <Link
      replace={false}
      style={
        Object {
          "textDecoration": "none",
        }
      }
      to="/places"
    >
      <WithStyles(Button)
        color="primary"
        variant="contained"
      >
        Explore to the List
      </WithStyles(Button)>
    </Link>
  </div>
</div>
`;
